snuba/utils/streams/dummy.py

```diff
@@ -1,58 +1,80 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from concurrent.futures import Future
 from datetime import datetime
 from typing import (
     Callable,
+    Generic,
     Mapping,
     MutableMapping,
     MutableSequence,
     Optional,
     Sequence,
+    Union,
 )
 
-from snuba.utils.streams.consumer import Consumer, ConsumerError
+from snuba.utils.streams.consumer import Consumer, ConsumerError, EndOfPartition
+from snuba.utils.streams.producer import MessageDetails, Producer
 from snuba.utils.streams.types import Message, Partition, Topic, TPayload
 
-
 epoch = datetime(2019, 12, 19)
 
 
+class DummyBroker(Generic[TPayload]):
+    def __init__(
+        self, topics: Mapping[Topic, Sequence[MutableSequence[TPayload]]]
+    ) -> None:
+        self.topics = topics
+        self.offsets: MutableMapping[str, MutableMapping[Partition, int]] = defaultdict(
+            dict
+        )
```

Comment:
Note that this doesn't do any sort of balanced assignment, I'd like to do that later to support testing rebalance behavior but it isn't necessary at the moment.